modules/programs/carapace.nix

```
@@ -61,18 +61,23 @@ in {
       };
     };
 
-    xdg.configFile =
-      mkIf (config.programs.fish.enable && cfg.enableFishIntegration) (
+    xdg.configFile = mkIf (config.programs.fish.enable
+      && cfg.enableFishIntegration
+      && lib.versionOlder pkgs.fish.version "4.0.0") (
```

Technically, you're checking the version of fish available in nixpkgs, not what the user has installed. 
```suggestion
      && lib.versionOlder config.programs.fish.package.version "4.0.0") (
```